File: src/binding/swift/Examples/one_minute.swift
// Copyright 2025-present the zvec project
import ZVec
import Foundation

@main
struct OneMinuteExample {
    static func main() {
        do {
            // 1. Create Schema
            let schema = Schema(name: "test_collection")
            try schema.addField(name: "vector", type: ZVEC_TYPE_VECTOR_FP32, dimension: 128)
            try schema.addField(name: "age", type: ZVEC_TYPE_INT32)

            // 2. Create and Open Collection
            let collection = try Collection(path: "/tmp/zvec_swift_example", schema: schema)

            // 3. Upsert Documents
            let doc1 = Doc()
            doc1.setPK("user_1")
            try doc1.setVector(field: "vector", vector: Array(repeating: 0.1, count: 128))
            try doc1.setInt32(field: "age", value: 30)

            try collection.upsert(docs: [doc1])
            try collection.flush()

            // 4. Get Stats
            let count = try collection.stats()
            print("Total docs: \(count)")

            // 5. Fetch Document
            let docs = try collection.fetch(pks: ["user_1"])
            for doc in docs {
                print("Fetched PK: \(doc.pk())")
            }
        } catch {
            print("Error: \(error)")
        }
    }
}

File: src/binding/go/examples/one_minute.go
// Copyright 2025-present the zvec project
package main

import (
	"fmt"
	"zvec"
)

func main() {
	// 1. Create Schema
	schema := zvec.NewSchema("test_collection")
	schema.AddField("vector", zvec.TypeVectorFp32, 128)
	schema.AddField("age", zvec.TypeInt32, 0)

	// 2. Create and Open Collection
	collection, err := zvec.CreateAndOpenCollection("/tmp/zvec_go_example", schema)
	if err != nil {
		panic(err)
	}

	// 3. Upsert Documents
	doc1 := zvec.NewDoc()
	doc1.SetPK("user_1")
	doc1.SetVector("vector", make([]float32, 128))
	doc1.SetInt32("age", 30)

	err = collection.Upsert([]*zvec.Doc{doc1})
	if err != nil {
		panic(err)
	}
	collection.Flush()

	// 4. Get Stats
	count, _ := collection.Stats()
	fmt.Printf("Total docs: %d\n", count)

	// 5. Fetch Document
	docs, err := collection.Fetch([]string{"user_1"})
	if err != nil {
		panic(err)
	}
	for _, d := range docs {
		fmt.Printf("Fetched PK: %s\n", d.PK())
	}
}
